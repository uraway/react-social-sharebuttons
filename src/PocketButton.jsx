// @flow
import React, { Component } from 'react';

type Props = {
  count?: 'horizontal' | 'vertical' | 'none',
};

type State = {
  initialized: boolean,
};

export default class PocketButton extends Component<Props, State> {
  node = null;

  static defaultProps = {
    count: '',
  };

  constructor(props: Props) {
    super(props);
    this.state = { initialized: false };
  }

  componentDidMount() {
    const { initialized } = this.state;
    if (initialized) {
      return;
    }

    const pocketbutton = this.node;
    const pocketscript = document.createElement('script');
    pocketscript.src = 'https://widgets.getpocket.com/v1/j/btn.js?v=1';
    pocketscript.async = true;
    pocketscript.id = 'pocket-btn-js';
    if (pocketbutton && pocketbutton.parentNode) pocketbutton.parentNode.appendChild(pocketscript);

    this.initialized();
  }

  initialized() {
    this.setState({ initialized: true });
  }

  render() {
    const { count, ...others } = this.props;
    return (
      <a
        ref={(node) => (this.node = node)}
        data-pocket-label="pocket"
        data-pocket-count={count}
        className="pocket-btn"
        data-lang="en"
        {...others}
      >
        Pocket
      </a>
    );
  }
}
