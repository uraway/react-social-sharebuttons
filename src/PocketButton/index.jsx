// @flow
import React, { Component } from 'react';

type Props = {
  count?: 'horizontal' | 'vertical' | 'none',
};

export default class PocketButton extends Component<Props> {
  node = null;

  static defaultProps = {
    count: '',
  };

  componentDidMount() {
    const s = document.createElement('script');
    s.src = 'https://widgets.getpocket.com/v1/j/btn.js?v=1';
    s.async = true;
    s.id = 'pocket-btn-js';
    if (this.node && this.node.parentNode) this.node.parentNode.appendChild(s);
  }

  render() {
    const { count, ...others } = this.props;
    return (
      <div>
        <a
          ref={(node) => (this.node = node)}
          data-pocket-label="pocket"
          data-pocket-count={count}
          className="pocket-btn"
          data-lang="en"
          {...others}
        >
          Pocket
        </a>
      </div>
    );
  }
}
