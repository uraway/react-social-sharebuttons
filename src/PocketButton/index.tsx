import { FC } from 'react';
import { useLayoutEffectOnce } from '../utils';

export type PocketButtonProps = {
  count?: 'horizontal' | 'vertical' | 'none';
};

const PocketButton: FC<PocketButtonProps> = (props) => {
  useLayoutEffectOnce(() => {
    const script = document.createElement('script');
    script.src = 'https://widgets.getpocket.com/v1/j/btn.js?v=1';
    script.async = true;
    script.id = 'pocket-btn-js';
    document.getElementsByTagName('head')[0]?.appendChild(script);

    return () => script.remove();
  });
  const { count, ...others } = props;
  return (
    <div>
      <a data-pocket-label="pocket" data-pocket-count={count} className="pocket-btn" data-lang="en" {...others}>
        Pocket
      </a>
    </div>
  );
};

PocketButton.defaultProps = {
  count: 'horizontal',
};

export default PocketButton;
