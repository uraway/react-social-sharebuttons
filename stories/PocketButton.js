import React from 'react';
import { select } from '@storybook/addon-knobs';
import { PocketButton } from '../dist/index.es.js'

export default {
  title: 'PocketButton',
};

const countOptions = {
  horizontal: 'horizontal',
  vertical: 'vertical',
  none: 'none',
};

export const Basic = () => <PocketButton count={select('count', countOptions, 'horizontal')} />;
