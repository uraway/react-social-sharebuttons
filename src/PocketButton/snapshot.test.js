import React from 'react';
import renderer from 'react-test-renderer';
import PocketButton from '.';

it('renders correctly with required props', () => {
  const tree = renderer.create(<PocketButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with other props', () => {
  const tree = renderer.create(<PocketButton url="vertical" />).toJSON();
  expect(tree).toMatchSnapshot();
});
