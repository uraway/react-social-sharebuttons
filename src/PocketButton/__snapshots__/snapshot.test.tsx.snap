// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with other props 1`] = `
<div>
  <a
    className="pocket-btn"
    data-lang="en"
    data-pocket-count="vertical"
    data-pocket-label="pocket"
  >
    Pocket
  </a>
</div>
`;

exports[`renders correctly with required props 1`] = `
<div>
  <a
    className="pocket-btn"
    data-lang="en"
    data-pocket-count="horizontal"
    data-pocket-label="pocket"
  >
    Pocket
  </a>
</div>
`;
